src/views/RegistrationView.vue

```diff
@@ -0,0 +1,105 @@
+<script setup>
+import { ref } from 'vue'
+import { useStore } from 'vuex'
+import { useRouter } from 'vue-router'
+import { fetchApiCall } from '@/utils/api';
+
+const store = useStore()
+const router = useRouter()
+const username = ref('')
+const password = ref('')
+const errorMessage = ref('')
+const showPassword = ref(false)
+
+// validation rule for vuetify components
+// we can add rules here in the future for passwords
+const rules = {
+    required: value => !!value || 'Required.'
+}
+
+const storeToken = (data) => {
+    const authToken = data.token
+    if (authToken) {
+        store.commit('setAuthToken', authToken)
+        return authToken
+    }
+}
+
+const handleError = (error) => {
+    console.error('API call failed with error:', error)
+    errorMessage.value = 'Failed to authenticate user'
+}
+
+const authenticateUser = async () => {
+    const requestBody = {
+        username: username.value,
+        password: password.value
+    }
+
+    const authHeaders = {
+        'Content-Type': 'application/json',
+        'Accept': 'application/json',
+    }
+
+    try {
+        await fetchApiCall({ url: store.state.observationPortalUrl + 'api-token-auth/', method: 'POST', body: requestBody, headers: authHeaders, successCallback: storeToken, failCallback: handleError })
+    } catch (error) {
+        handleError(error)
+    }
+}
+
+const storeUser = (user) => {
+    store.commit('setUsername', username.value)
+    store.commit('setUserProfile', user)
+    store.commit('setProjects', user.proposals)
+    router.push({ name: 'ProjectView' })
+}
+
+const getUserProfile = async () => {
+    await authenticateUser ()
+    const token = store.state.authToken
+    const authHeaders = {
+        'Content-Type': 'application/json',
+        'Accept': 'application/json',
+        'Authorization': token ? `Token ${token}` : null,
```

Comment:
I think this is unneccessary because token will either be empty string or a value you set it to, so either way you will fail authentication on the request if its empty string or just empty...

Reply:
true, updated!